File: Programmers/자바기초/정수를나선형으로배치.java
class Solution {
    public static int[][] solution(int n) {
        int[][] answer = new int[n][n];
        int num = 0;
        int i=0;
        int j=0;
        String direction = "right";

        while (num < n*n) {
            
            num++;
            if (direction.equals("right")) {                
                answer[i][j] = num;
                j++;
                if (j == n || answer[i][j] != 0) {
                    j--;
                    i++;
                    direction = "down";
                    continue;
                }
                
            }
            
            else if (direction.equals("down")) {        
                answer[i][j] = num;
                i++;
                if (i == n || answer[i][j] != 0) {
                    i--;
                    j--;
                    direction = "left";
                    continue;
                }
            }
            
            else if (direction.equals("left")) {
                answer[i][j] = num;
                j--;
                if (j < 0 || answer[i][j] != 0) {
                    j++;
                    i--;
                    direction = "up";
                    continue;
                }
            }
            
            else if (direction.equals("up")) {
                answer[i][j] = num;
                i--;
                if (i < 0 || answer[i][j] != 0) {
                    i++;
                    j++;
                    direction = "right";
                    continue;
                }
                
            }

            else;
            
        }

        return answer;
    }
}

File: Programmers/제로베이스_문제추천_py/레벨0/정수를나선형으로배치.py
def solution(n):
    answer = [[]]
    board = []
    direction = "right"
    num = 0
    for _ in range(n) :
        board.append([0]*n)
    
    i = 0
    j = 0
    while num < n*n :
        num += 1
        
        if direction == "right" :
            board[i][j] = num
            j += 1
            if j >= n or board[i][j] != 0 :
                direction = "down"
                i += 1
                j -= 1
                continue
                
        
        if direction == "down" :
            board[i][j] = num
            i += 1
            if i >= n or board[i][j] != 0 :
                direction = "left"
                i -= 1
                j -= 1
                continue
        
        if direction == "left" :
            board[i][j] = num
            j -= 1
            if j < 0 or board[i][j] != 0 :
                direction = "up"
                j += 1
                i -= 1
                continue
        
        if direction == "up" :
            board[i][j] = num
            i -= 1
            if i < 0 or board[i][j] != 0 :
                direction = "right"
                i += 1
                j += 1
                continue
        

    
    answer = board
    return answer